### PR TITLE
Fix: add max number of fixes per rule

### PR DESCRIPTION
**Pull request by henry-lp**
This PR enable the user to specify the number of rules to be fixed. 

issue #112

@monperrus , @fermadeiral

### CODE REVIEW

File: src/main/java/sorald/Constants.java [hunk 1]
```diff
@@ -11,7 +11,8 @@
 	public static final String ARG_GIT_REPO_PATH = "gitRepoPath";
 	public static final String ARG_PRETTY_PRINTING_STRATEGY = "prettyPrintingStrategy";
 	public static final String ARG_FILE_OUTPUT_STRATEGY = "fileOutputStrategy";
-
+	public static final String ARG_MAX_FIXES_PER_RULES = "maxFilesPerRule";
```
Comment (fermadeiral) [inline src/main/java/sorald/Constants.java:14]: `ARG_MAX_FIXES_PER_RULES` -> `ARG_MAX_FIXES_PER_RULE`

`maxFilesPerRule` -> `maxFixesPerRule`

File: src/main/java/sorald/processor/SoraldAbstractProcessor.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
 /** superclass for all processors */
 public abstract class SoraldAbstractProcessor<E extends CtElement> extends AbstractProcessor<E> {
 	private Set<Bug> bugs;
+	private int maxFixesNbs = Integer.MAX_VALUE;
```
Comment (fermadeiral) [inline src/main/java/sorald/processor/SoraldAbstractProcessor.java:25]: `maxFixesNbs` -> `maxFixes`

File: src/main/java/sorald/processor/SoraldAbstractProcessor.java [hunk 3]
```diff
@@ -22,6 +22,8 @@
...
+	private int currentFixesNbs = 0;
```
Comment (fermadeiral) [inline src/main/java/sorald/processor/SoraldAbstractProcessor.java:26]: `currentFixesNbs` -> `nbFixes`

File: src/main/java/sorald/processor/SoraldAbstractProcessor.java [hunk 4]
```diff
@@ -48,6 +50,19 @@
 		}
 	}
 
+	public SoraldAbstractProcessor setMaxFixesNbs(int maxFixesNbs) {
```
Comment (fermadeiral) [inline src/main/java/sorald/processor/SoraldAbstractProcessor.java:53]: `setMaxFixesNbs(int maxFixesNbs)` -> `setMaxFixes(int maxFixes)`

File: src/main/java/sorald/processor/SoraldAbstractProcessor.java [hunk 5]
```diff
@@ -48,6 +50,19 @@
...
+		this.maxFixesNbs = maxFixesNbs;
+		return this;
+	}
+
+	public int getCurrentFixesNbs() {
```
Comment (fermadeiral) [inline src/main/java/sorald/processor/SoraldAbstractProcessor.java:58]: `getCurrentFixesNbs` -> `getNbFixes`

**Comment by fermadeiral** [comment]
You should also update the README.

File: src/main/java/sorald/Constants.java (hunk 1 above)
Comment (henry-lp) [reply src/main/java/sorald/Constants.java:14]: done :)

File: README.md [hunk 6]
```diff
@@ -71,6 +71,10 @@ The arguments are the following:
         files, including the unchanged ones, will be created in the workspace.
         (default: CHANGED_ONLY)
 
+  [--maxFixesPerRule <maxFixesPerRule>]
+        Mode for outputting files: Max number of fixes per rule.
+        (default: Integer.MAX_VALUE )
+
```
Comment (fermadeiral) [inline README.md:77]: This doesn't match what is actually going to be shown to the user. To get the correct description, the best way is to build the jar and to call it by passing "-h".

File: src/main/java/sorald/processor/SoraldAbstractProcessor.java (hunk 2 above)
Comment (fermadeiral) [inline src/main/java/sorald/processor/SoraldAbstractProcessor.java:25]: `maxFixesNbs` -> `maxFixes`

File: src/main/java/sorald/processor/SoraldAbstractProcessor.java [hunk 7]
```diff
@@ -48,6 +50,19 @@
 		}
 	}
 
+	public SoraldAbstractProcessor setMaxFixes(int maxFixesNbs) {
```
Comment (fermadeiral) [inline src/main/java/sorald/processor/SoraldAbstractProcessor.java:53]: `maxFixesNbs` -> `maxFixes`

File: src/main/java/sorald/processor/SoraldAbstractProcessor.java [hunk 8]
```diff
@@ -48,6 +50,19 @@
...
+		this.maxFixesNbs = maxFixesNbs;
+		return this;
+	}
+
+	public int getnbFixes() {
```
Comment (fermadeiral) [inline src/main/java/sorald/processor/SoraldAbstractProcessor.java:58]: `getnbFixes` -> `getNbFixes`

File: README.md (hunk 6 above)
Comment (henry-lp) [reply README.md:77]: done

**Comment by henry-lp** [comment]
Ready for review :) .

**Comment by fermadeiral** [comment]
Thank you, Henry!

**Pull request merged**